tests/unit/Parameters/ConfigurationValidatorTest.php

```diff
@@ -28,90 +28,128 @@
 
 use PHPUnit\Framework\TestCase;
 use PrestaShop\Module\AutoUpgrade\Parameters\ConfigurationValidator;
-use UnexpectedValueException;
+use PrestaShop\Module\AutoUpgrade\UpgradeTools\Translator;
 
 class ConfigurationValidatorTest extends TestCase
 {
+    /** @var ConfigurationValidator */
+    private $validator;
+
+    protected function setUp()
+    {
+        $translator = $this->createMock(Translator::class);
+
+        $translator->method('trans')->willReturnCallback(function ($string, $params) {
+            return sprintf($string, ...$params);
+        });
+
+        $this->validator = new ConfigurationValidator($translator);
+    }
+
     public function testValidateChannelSuccess()
     {
-        $validator = new ConfigurationValidator();
+        $result = $this->validator->validate(['channel' => 'online']);
+        $this->assertEmpty($result);
 
-        $validator->validate(['channel' => 'online']);
-        $validator->validate(['channel' => 'local']);
+        $result = $this->validator->validate(['channel' => 'local']);
+        $this->assertEmpty($result);
     }
 
     public function testValidateChannelFail()
     {
-        $validator = new ConfigurationValidator();
-
-        $this->expectException(UnexpectedValueException::class);
-        $this->expectExceptionMessage('Unknown channel toto');
+        $channel = 'toto';
 
-        $validator->validate(['channel' => 'toto']);
+        $result = $this->validator->validate(['channel' => $channel]);
+        $this->assertEquals([
+            'message' => 'Unknown channel ' . $channel,
+            'target' => 'channel',
+        ], $result[0]);
```

Comment:
If that is relevant, you can test the whole array returned

```suggestion
        $this->assertEquals([
            [
                'message' => 'Unknown channel toto',
                'target' => 'channel',
            ]
        ], $result);
```

* You make sure there is no other unexpected errors
* You avoid unnecessary logic in the test (https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html)